Sources/BottomSheetCalculator.swift

```diff
@@ -30,68 +30,98 @@ struct BottomSheetCalculator {
-        currentTargetOffsetIndex: Int
-    ) -> TranslationState? {
-        guard currentTargetOffsetIndex >= 0 && currentTargetOffsetIndex < targetOffsets.count else { return nil }
-        guard thresholds.count == targetOffsets.count + 1 else { return nil }
-
-        let currentTargetOffset = targetOffsets[currentTargetOffsetIndex]
-        let lowerBound = currentTargetOffset - thresholds[currentTargetOffsetIndex]
-        let upperBound = currentTargetOffset + thresholds[currentTargetOffsetIndex + 1]
-        let currentArea = lowerBound ... upperBound
-
-        if currentArea.contains(destination) {
-            // Within the area of the current target offset, allow dragging.
-            return TranslationState(nextOffset: destination, targetOffset: currentTargetOffset, isDismissible: false)
-        } else if destination < currentTargetOffset {
-            let targetOffset = targetOffsets.first(where: { $0 < destination })
-            // Above the area of the current target offset, allow dragging if the next target offset is found.
-            return TranslationState(
-                nextOffset: targetOffset == nil ? source : destination,
-                targetOffset: targetOffset ?? currentTargetOffset,
-                isDismissible: false
-            )
-        } else {
-            let targetOffset = targetOffsets.first(where: { $0 > destination })
-            // Below the area of the current target offset,
-            // allow dragging and set as dismissable if the next target offset is not found.
-            return TranslationState(
-                nextOffset: destination,
-                targetOffset: targetOffset ?? currentTargetOffset,
-                isDismissible: targetOffset == nil
+        // Calculate lower bounds
+        let lowerOffsets = targetOffsets[currentTargetIndex...]
+        let lowerThresholds = thresholds[(currentTargetIndex + 1)...]
+        let lowerBounds = zip(lowerOffsets, lowerThresholds).map { (offset, threshold) -> CGFloat in
+            offset - threshold
+        }
+
+        // Calculate upper bounds
+        let upperOffsets = targetOffsets[...currentTargetIndex]
+        let upperThresholds = thresholds[...currentTargetIndex]
+        let upperBounds = zip(upperOffsets, upperThresholds).map { (offset, threshold) -> CGFloat in
+            offset + threshold
+        }
+
+        let bounds = upperBounds.dropFirst() + lowerBounds
+        var upperBound = upperBounds.first ?? 0
+
+        // Model used to control offsets bigger than or equal to maxOffset
+        let bottomModel = LimitModel(
+            targetOffset: maxOffset,
+            isDismissible: isDismissible,
+            compare: >=
+        )
+
+        var models: [BottomSheetModel] = [bottomModel]
+
+        for (index, lowerBound) in bounds.enumerated() {
+            let model = RangeModel(
+                targetOffset: targetOffsets[index],
+                range: lowerBound ..< upperBound,
```

Comment:
The demo app crashes on this line with `Fatal error: Can't form Range with upperBound < lowerBound` if you swap target heights in `BottomSheetTransitioningDelegate(targetHeights: [UIScreen.main.bounds.size.height - 200, .bottomSheetAutomatic])`. I guess it's due to offsets not being sorted anymore.